addons/mrp/wizard/mrp_product_produce.py

```diff
@@ -118,6 +120,37 @@ def do_produce(self):
             })
         return {'type': 'ir.actions.act_window_close'}
 
+    def _set_quantity_done(self, move, qty):
+        """
+        Set the given quantity as quantity done on the move through the move lines. The method is
+        able to handle move lines with a different UoM than the move (but honestly, this would be
+        looking for trouble...).
+        @param move: stock.move
+        @param qty: quantity in the UoM of move.product_uom
+        """
+        for ml in move.move_line_ids:
+            # Convert move line qty into move uom
+            ml_qty = ml.product_uom_qty
```

Comment:
You should remove the qty_done since the user could produce multiple time and ml could already have some quantity done on it.